giskard/registry/giskard_test.py

```diff
@@ -102,6 +102,20 @@ def load(cls, local_dir: Path, uuid: str, meta: TestFunctionMeta):
     def get_builder(self):
         return type(self)
 
+    def assert_(self):
+        """Wrap execution of the test into a "assert" for unit test executions."""
+        result = self.execute()
+        if isinstance(result, bool):
+            assert result
+        else:
+            if result.messages:
+                message = " ".join([repr(message) for message in result.messages])
+            else:
+                # Pass more context in case the message is empty
+                message = " ".join(str(result).replace("\n", " ").split())
+
+            assert result.passed, message
```

Comment:
Might be worth to add a `__bool__` and enrich the `__repr__` value in the `TestResult` to simplify this to something like:

```python
result = self.execute()
assert result, message or result
```

Reply:
I have yet to deep dive into the `message` attribute but it seems to be `None` for many tests and it might be worth "enriching" this attribute for more meaning full test reporting.

Reply:
Indeed, maybe another property like test summary that contains a one-liner about the result.
It also seems that `message` is quite a riche data type